### PR TITLE
add remaining host functions to wamr-host

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds the last set of host functions to `wamr-host` - the last step of #149 

### Context of Change

#149

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Test Plan

CI passes.

### CODE REVIEW

File: projects/its/codecov_tests/src/host_bindings_loose_for_testing.rs [hunk 1]
```diff
@@ -31,6 +31,19 @@ pub fn account_keylet(
     32
 }
 
+#[allow(unused)]
+#[allow(clippy::missing_safety_doc)]
+pub fn amm_keylet(
+    issue1_ptr: *const u8,
+    issue1_len: usize,
+    issue2_ptr: i32,
+    issue2_len: i32,
```
[the remark this answers is not in the document]
Comment (mvadari) [reply projects/its/codecov_tests/src/host_bindings_loose_for_testing.rs:40]: This is on purpose, to test edge cases
